code/modules/projectiles/guns/rifles.dm

```diff
@@ -630,19 +632,10 @@
 	recalculate_attachment_bonuses()
 	if(iff_enabled)
 		add_bullet_trait(BULLET_TRAIT_ENTRY_ID("iff", /datum/element/bullet_trait_iff))
+		AddComponent(/datum/component/iff_fire_prevention)
 	else
 		remove_bullet_trait("iff")
-
-/obj/item/weapon/gun/rifle/m46c/recalculate_attachment_bonuses()
-	. = ..()
-	if(iff_enabled)
-		modify_fire_delay(FIRE_DELAY_TIER_12)
-		remove_firemode(GUN_FIREMODE_BURSTFIRE)
-		remove_firemode(GUN_FIREMODE_AUTOMATIC)
-
-	else
-		add_firemode(GUN_FIREMODE_BURSTFIRE)
-		add_firemode(GUN_FIREMODE_AUTOMATIC)
+		GetExactComponent(/datum/component/iff_fire_prevention).RemoveComponent()
```

Comment:
i think you should rather have the component "pause" with a simple variable, you're not really supposed to use GetComponent

Reply:
And how do I pass a variable to a component without getting it? Can you give an example?

Reply:
Probably I can make a new signal but eh...